pennylane/templates/state_preparations/mottonen.py

```diff
@@ -120,8 +120,8 @@ def _uniform_rotation_dagger(gate, alpha, control_wires, target_wire):
     gray_code_rank = len(control_wires)
 
     if gray_code_rank == 0:
-        if theta[0] != 0.0:
-            gate(theta[0], wires=[target_wire])
+        if qml.math.all(theta[..., 0] != 0.0):  # only omit gate if angle zero in all batches?
```

Comment:
Right, there was an optimization here that omitted the rotation gate if the angle was zero. Are we ok with the gate only being removed if the entire batch of angles is zero?

```suggestion
        if qml.math.all(theta[..., 0] != 0.0):
```

Reply:
What a good question! Seen this way, working with batches limits quite a lot the optimization of the circuits. What do you think about this @antalszava , do you see any way to optimize this?

Reply:
Just saw this comment, my bad! I think this is the best approach we have for now :thinking: Good thing to note.

An alternative that comes to mind is to add logic to basic rotation gates to handle cases when they are applied with angle zero. This might be something the user intentionally does, however. Not a common use case, but could potentially arise e.g., to introduce noise in the simulations.